codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java

```diff
@@ -235,6 +237,11 @@ public class AddAwsConfigFields implements GoIntegration {
                     .documentation("The inclusive min request body size to be compressed.")
                     .servicePredicate(RequestCompression::isRequestCompressionService)
                     .generatedOnClient(false)
+                    .build(),
+            AwsConfigField.builder()
+                    .name(SDK_AID_ENDPOINT_MODE)
+                    .type(getAwsAccountIDModeSymbol("AIDMode"))
```

Comment:
Add and use `SdkGoTypes.Aws.AccountIDEndpointMode` (once you've moved the enum there) so you can use that directly.